### PR TITLE
Etfe 2491 - add the sad section importnumber/ add / remove form list page

### PR DESCRIPTION
**Pull request by funkmaster20th**
plus indexing of the import number

### CODE REVIEW

**Comment by funkmaster20th** [comment]
Screenshot 2023-11-06 at 18 33 48

File: app/controllers/sections/sad/SadAddToListController.scala [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+package controllers.sections.sad
+
+import controllers.actions._
+import forms.sections.sad.SadAddToListFormProvider
+import models.NormalMode
+import models.requests.DataRequest
+import models.sections.sad.SadAddToListModel
+import navigation.SadNavigator
+import pages.sections.sad.{SadAddToListPage, SadSection, SadsSection}
```
Comment (joenjones07) [inline app/controllers/sections/sad/SadAddToListController.scala:25]: Unused Import SadSection

File: test/controllers/sections/sad/ImportNumberControllerSpec.scala [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.sections.sad
+
+import base.SpecBase
+import forms.sections.sad.ImportNumberFormProvider
+import mocks.services.MockUserAnswersService
+import models.{NormalMode, UserAnswers}
```
Comment (joenjones07) [inline test/controllers/sections/sad/ImportNumberControllerSpec.scala:22]: Unused Import UserAnswers

File: test/controllers/sections/sad/SadAddToListControllerSpec.scala [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+ */
+
+package controllers.sections.sad
+
+import base.SpecBase
+import forms.sections.sad.SadAddToListFormProvider
+import mocks.services.MockUserAnswersService
+import models.{NormalMode, UserAnswers}
+import models.sections.sad.SadAddToListModel
+import models.sections.sad.SadAddToListModel.{NoMoreToCome, Yes}
```
Comment (joenjones07) [inline test/controllers/sections/sad/SadAddToListControllerSpec.scala:24]: Unused Import

File: test/forms/sections/sad/ImportNumberFormProviderSpec.scala [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms.sections.sad
+
+import fixtures.messages.sections.sad.ImportNumberMessages
+import forms.XSS_REGEX
+import forms.ALPHANUMERIC_REGEX
```
Comment (joenjones07) [inline test/forms/sections/sad/ImportNumberFormProviderSpec.scala:21]: Unused Import

File: app/forms/sections/sad/ImportNumberFormProvider.scala [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms.sections.sad
+
+import forms.{ONLY_ALPHANUMERIC_REGEX, XSS_REGEX}
```
Comment (joenjones07) [inline app/forms/sections/sad/ImportNumberFormProvider.scala:19]: Unused Import XSS_REGEX

File: test/controllers/sections/sad/SadAddToListControllerSpec.scala [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.sections.sad
+
+import base.SpecBase
+import forms.sections.sad.SadAddToListFormProvider
+import mocks.services.MockUserAnswersService
+import models.{NormalMode, UserAnswers}
```
Comment (joenjones07) [inline test/controllers/sections/sad/SadAddToListControllerSpec.scala:22]: Unused Import UserAnswers

File: test/views/sections/sad/SadAddToListViewSpec.scala [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+ */
+
+package views.sections.sad
+
+import base.ViewSpecBase
+import fixtures.messages.sections.sad.SadAddToListMessages
+import forms.sections.sad.SadAddToListFormProvider
+import models.NormalMode
+import models.requests.DataRequest
+import models.sections.sad.SadAddToListModel
```
Comment (joenjones07) [inline test/views/sections/sad/SadAddToListViewSpec.scala:24]: Unused Import

Comment (funkmaster20th) [reply test/views/sections/sad/SadAddToListViewSpec.scala]: removed

File: test/controllers/sections/sad/SadAddToListControllerSpec.scala (hunk 6 above)
Comment (funkmaster20th) [reply test/controllers/sections/sad/SadAddToListControllerSpec.scala]: removed

File: app/forms/sections/sad/ImportNumberFormProvider.scala (hunk 5 above)
Comment (funkmaster20th) [reply app/forms/sections/sad/ImportNumberFormProvider.scala]: removed

File: test/forms/sections/sad/ImportNumberFormProviderSpec.scala (hunk 4 above)
Comment (funkmaster20th) [reply test/forms/sections/sad/ImportNumberFormProviderSpec.scala]: removed

File: test/controllers/sections/sad/SadAddToListControllerSpec.scala (hunk 3 above)
Comment (funkmaster20th) [reply test/controllers/sections/sad/SadAddToListControllerSpec.scala]: removed

File: test/controllers/sections/sad/ImportNumberControllerSpec.scala (hunk 2 above)
Comment (funkmaster20th) [reply test/controllers/sections/sad/ImportNumberControllerSpec.scala]: removed

File: app/controllers/sections/sad/SadAddToListController.scala (hunk 1 above)
Comment (funkmaster20th) [reply app/controllers/sections/sad/SadAddToListController.scala]: removed

**Review by joenjones07 — APPROVED** [review]

**Pull request merged**